tools/convert_datasets/otb100/download_otb100.py

```diff
@@ -0,0 +1,65 @@
+import argparse
+import multiprocessing
+import os
+import os.path as osp
+import re
+import socket
+from urllib import error, request
+
+from bs4 import BeautifulSoup
+from tqdm import tqdm
+
+
+def download_url(url_savedir_tuple):
+    url = url_savedir_tuple[0]
+    saved_dir = url_savedir_tuple[1]
+    video_zip = osp.basename(url)
+    if not osp.isdir(saved_dir):
+        os.makedirs(saved_dir)
+    saved_file = saved_dir + '/' + video_zip
+
+    try:
+        request.urlretrieve(url, saved_file)
+    except error.HTTPError as e:
+        print(e)
+        print('\r\n' + url + ' download failed!' + '\r\n')
+    except socket.timeout:
+        count = 1
+        while count <= 5:
+            try:
+                request.urlretrieve(url, saved_file)
+                break
+            except socket.timeout:
+                err_info = 'Reloading %s for %d time' % (video_zip, count)
+                print(err_info)
+                count += 1
+        if count > 5:
+            print('downloading %s failed!' % video_zip)
+
+
+def parse_url(homepage, href=None):
+    html = request.urlopen(homepage + 'datasets.html').read().decode('utf-8')
+    soup = BeautifulSoup(html, features='html.parser')
```

Comment:
Do we have alternatives for `BeautifulSoup`?

Reply:
we also decode it manualy by regular expression, but BeautifulSoup is more neat, flexible and general for web crawling.